cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/DeleteKafkaClusterOperation.java

```diff
@@ -45,7 +50,27 @@ public void execute(Vertx vertx, K8SUtils k8s, Handler<AsyncResult<Void>> handle
                 Future<Void> futureStatefulSet = Future.future();
                 OperationExecutor.getInstance().execute(new DeleteStatefulSetOperation(namespace, name), futureStatefulSet.completer());
 
-                CompositeFuture.join(futureConfigMap, futureService, futureHeadlessService, futureStatefulSet).setHandler(ar -> {
+                Future<Void> futurePersistentVolumeClaim = Future.future();
+                if ((kafka.getStorage().type() == Storage.StorageType.PERSISTENT_CLAIM) && kafka.getStorage().isDeleteClaim()) {
+
+                    List<Future> futurePersistentVolumeClaims = new ArrayList<>();
+                    for (int i = 0; i < kafka.getReplicas(); i++) {
+                        Future<Void> f = Future.future();
+                        futurePersistentVolumeClaims.add(f);
+                        OperationExecutor.getInstance().execute(new DeletePersistentVolumeClaimOperation(namespace, "kafka-storage-" + name + "-" + i), f.completer());
+                    }
+                    CompositeFuture.join(futurePersistentVolumeClaims).setHandler(ar -> {
+                        if (ar.succeeded()) {
+                            handler.handle(Future.succeededFuture());
+                        } else {
+                            handler.handle(Future.failedFuture("Failed to delete persistent volume claims"));
+                        }
+                    });
+                } else {
+                    futurePersistentVolumeClaim.complete();
+                }
+
+                CompositeFuture.join(futureConfigMap, futureService, futureHeadlessService, futureStatefulSet, futurePersistentVolumeClaim).setHandler(ar -> {
```

Comment:
You are basically firing all things at the same time. That can mean that you can try to delete the PVCs while still used by the StatefulSet - could that cause some problems? Can you delete the PVCs used by a running Pod?

Reply:
Good point. Up to know I had no problem but I'm going to test deeply. The point is that in any case the fabric8 API are "asynchronous" so if we want to be sure that all the pods are gone before deleting PVCs we have to introduce a different internal API (in the K8S Utils ?) in order to "watch" such operation. Without this "sync" all the fabric8 operations we'll start are async and deleting a PVC is much faster than deleting a pod.

Reply:
Ok, that is a valid point that even that the deleteStatefulSet future ends doesn't mean that the stateful set is gone. In that case probably doesn't make much sense to do anything special for the PVCs.

Reply:
I have just tried to delete "manually" `oc delete pvc' command a PVC for a pod. OpenShift doesn't complain about that and the PVC is deleted without any problems. Interesting is that thanks to the caching (?) the Kafka broker which was using such PVC continues to work "storing" messages and delivering them to consumers. Of course, If I delete the pod, it will be restarted and a new PVC is created bounded to a new PV so previous messages are gone (so a new connected clients won't get messages "stored" previously on the deleted PVC + pod).

Reply:
Ok ... It is good that Kafka doesn't crash immediately, because that might trigger the pod restart and create new PVC. But it sounds like this will work fine.